### PR TITLE
Alerting: Update legacy migration docs to include Upgrade Preview

### PR DESCRIPTION
**Pull request by JacobsonMT**
**What is this feature?**

This PR is a general restructuring of the `Upgrade alerting` docs page to better focus on the upgrade aspect instead of the opt-in/opt-out aspect. In addition, it adds the new method of `Upgrade with Preview` added here: https://github.com/grafana/grafana/pull/80071

**Why do we need this feature?**

Docs need to be updated to include `Upgrade with Preview` and at the same time the introduction of a choice between methods necessitated a restructuring that better outlines their differences.

**Who is this feature for?**

Legacy alerting users looking to upgrade to Grafana Alerting.

**Special notes for your reviewer:**

- The names of `Simple Upgrade` and `Upgrade with Preview` are not set it stone and can be changed.
- Should there be screenshots anywhere on this page?

Screenshot from 2024-01-16 09-37-09

### CODE REVIEW

File: docs/sources/alerting/set-up/migrating-alerts/_index.md [hunk 1]
```diff
@@ -13,85 +13,133 @@ weight: 150
 
 # Upgrade Alerting
 
-Grafana Alerting is enabled by default for new installations or existing installations whether or not legacy alerting is configured.
-
 {{% admonition type="note" %}}
-When upgrading, your dashboard alerts are migrated to a new format. This migration can be rolled back easily by opting out. If you have any questions regarding this migration, please contact us.
+Legacy alerting will be removed in Grafana v11.0.0, which is scheduled for release in April 2024. We recommend that you upgrade to Grafana Alerting as soon as possible.
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/migrating-alerts/_index.md:17]: I think we should just say the version - but not promise a date here

File: docs/sources/alerting/set-up/migrating-alerts/_index.md [hunk 2]
```diff
@@ -13,85 +13,133 @@ weight: 150
...
-[alerting]
-enabled = true
+{{% admonition type="note" %}}
+Alerts generated by the new alerting system will be visible in the **Alerting** section of the navigation panel but will not be active until the upgrade is finalized.
+{{% /admonition %}}
 
-[unified_alerting]
-enabled = false
-```
+### Detailed Steps
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/migrating-alerts/_index.md:70]:
```suggestion
To upgrade with preview, complete the following steps.
```

File: docs/sources/alerting/set-up/migrating-alerts/_index.md [hunk 3]
```diff
@@ -13,85 +13,133 @@ weight: 150
...
 
 {{% admonition type="note" %}}
-This topic is only relevant for OSS and Enterprise customers. Contact customer support to enable or disable Grafana Alerting for your Grafana Cloud stack.
+Any errors encountered during the upgrade process will fail the upgrade and prevent Grafana from starting. If this occurs, you can roll back to legacy alerting by disabling Grafana Alerting and re-enabling legacy alerting.
 {{% /admonition %}}
 
-The `ngalert` toggle previously used to enable or disable Grafana Alerting is no longer available.
-
-## Deactivate alerting
+### Detailed Steps
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/migrating-alerts/_index.md:113]:
```suggestion
To perform the simple upgrade, complete the following steps.
```

**Comment by JacobsonMT** [comment]
Thanks for the review @brendamuir!

>hey matt - first up two structural questions/comments - could we move "opt out" further down to near the "opt in" section - and then introduce the two upgrade options straight after the table perhaps with a new heading "Upgrade options". Then further down there is a rolling back and a roll back section - are those the same - in which case can we merge them? or if different - can we rename, so it isn't confusing? will review the rest inline :)

I'm a bit confused, this new version doesn't have an "opt out", "opt in", or "roll back" section. Might be a render issue somewhere?

File: docs/sources/alerting/set-up/migrating-alerts/_index.md [hunk 4]
```diff
@@ -13,85 +13,140 @@ weight: 150
...
-enabled = true
+While we recommend the **Upgrade with Preview** method for its enhanced safety and control, the **Simple Upgrade Method** exists for specific situations where a preview environment is not essential. For example, if you have a large fleet of Grafana instances and want to upgrade them all without the need to review and adjust each one individually.
 
-[unified_alerting]
-enabled = false
-```
+Configure your Grafana instance to enable Grafana Alerting and disable legacy alerting. Then restart Grafana to automatically migrate your existing alert rules and notification channels to the new Grafana Alerting system.
 
-Installations that have been migrated to Grafana Alerting can roll back to legacy alerting at any time.
+Once Grafana Alerting is enabled, you can now review and adjust your upgraded alerts in the **Alerting** section of the navigation panel as well as export them for as-code setup.
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/migrating-alerts/_index.md:112]:
```suggestion
Once Grafana Alerting is enabled, you can review and adjust your upgraded alerts in the **Alerting** section of the navigation panel as well as export them for as-code setup.
```

**Review by brendamuir — APPROVED** [review]

**Pull request merged**